application/src/main/java/org/togetherjava/tjbot/commands/moderation/temp/RevocableRoleBasedAction.java

```diff
@@ -0,0 +1,55 @@
+package org.togetherjava.tjbot.commands.moderation.temp;
+
+import net.dv8tion.jda.api.exceptions.ErrorResponseException;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Role based moderation actions that can be revoked, for example a {@link TemporaryMuteAction} or a
+ * {@link TemporaryBanAction}, which are applied implicitly purely by the presence of a role.
+ */
+abstract class RevocableRoleBasedAction implements RevocableModerationAction {
+    private static final Logger logger = LoggerFactory.getLogger(RevocableRoleBasedAction.class);
+
+    private final String actionName;
+
+    /**
+     * Creates a new role based action.
+     *
+     * @param actionName the action name to be used in logging in case of a failure, e.g.
+     *        {@code "mute"}, {@code "quarantine"}
+     */
+    RevocableRoleBasedAction(@NotNull String actionName) {
+        this.actionName = actionName;
+    }
+
+    @Override
+    public @NotNull FailureIdentification handleRevokeFailure(@NotNull Throwable failure,
+            long targetId) {
+
+        if (failure instanceof ErrorResponseException errorResponseException) {
+            switch (errorResponseException.getErrorResponse()) {
+                case UNKNOWN_USER -> logger.debug(
+                        "Attempted to revoke a temporary {} but user '{}' does not exist anymore.",
+                        actionName, targetId);
+                case UNKNOWN_MEMBER -> logger.debug(
+                        "Attempted to revoke a temporary {} but user '{}' is not a member of the guild anymore.",
+                        actionName, targetId);
+                case UNKNOWN_ROLE -> logger.warn(
+                        "Attempted to revoke a temporary {} but the {} role can not be found.",
+                        actionName, actionName);
+                case MISSING_PERMISSIONS -> logger.warn(
+                        "Attempted to revoke a temporary {} but the bot lacks permission.",
+                        actionName);
+                default -> {
+                    return FailureIdentification.UNKNOWN;
+                }
```

Comment:
Sonar correctly complains about the unecessary `{...}` here.

Reply:
on a `return`? I believe a `return` requires the block, as it won't compile without it.

Reply:
return is implicit for one-liners. just `default -> FailureIdentification.UNKNOWN;`

Reply:
then dont directly return out of it but assign some variable so that both flow into the same return statement.
```java
FailureIdentification identification = FailureIdentification.KNOWN;
switch (...) {
  ...
  default -> identification = FailureIdentification.UNKNOWN;
}

return identification;
```

Dunno, try around, maybe you find something nicer. But I want to avoid merging something that introduces Sonar flags.

Reply:
Sonar is wrong, like usually.

Reply:
In this case, it is. We can click on "ignore", thats not an issue. But I still think that we could improve here, as reported above.

Reply:
Finally I can say it! fuck you Sonar \:)

And I'm fine with changing it so 👍

Reply:
This is my other idea, but that will cause another sonar violation.
```java
FailureIdentification handlingResult = FailureIdentification.UNKNOWN;

if (failure instanceof ErrorResponseException errorResponseException) {
    handlingResult = FailureIdentification.KNOWN;

    switch (errorResponseException.getErrorResponse()) {
        case UNKNOWN_USER -> logger.debug(
                "Attempted to revoke a temporary {} but user '{}' does not exist anymore.",
                actionName, targetId);
        case UNKNOWN_MEMBER -> logger.debug(
                "Attempted to revoke a temporary {} but user '{}' is not a member of the guild anymore.",
                actionName, targetId);
        case UNKNOWN_ROLE -> logger.warn(
                "Attempted to revoke a temporary {} but the {} role can not be found.",
                actionName, actionName);
        case MISSING_PERMISSIONS -> logger.warn(
                "Attempted to revoke a temporary {} but the bot lacks permission.",
                actionName);
        default -> handlingResult = FailureIdentification.UNKNOWN;
    }
}

return handlingResult;
```
```
Assignments should not be made from within sub-expressions

Assignments within sub-expressions are hard to spot and therefore make the code less readable.
Ideally, sub-expressions should not have side-effects.
```

I think the best solution is to `ignore` this one.

Reply:
Can you double check whether `default -> return foo;` actually works without curlies? just to make sure.

Reply:
That said, I marked it as false-positive:

![sonar](https://i.imgur.com/AGJ6aMj.png)